### PR TITLE
Initial release.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Comment by floitsch** [comment]
No tests and examples yet.

I will add them a bit later.

File: src/pipe.toit [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+      state_.clear_state READ_EVENT_
+
+  write x from = 0 to = x.size:
+    state_.wait_for_state WRITE_EVENT_ | ERROR_EVENT_
+    bytes_written := write_primitive_ resource_ x from to
+    if bytes_written == 0: state_.clear_state WRITE_EVENT_
+    return bytes_written
+
+  close:
+    close_ resource_
```
Comment (kasperl) [inline src/pipe.toit:81]: Be a sport and dispose the resource state too.

```
if state_:
  state_.dispose
  state_  = null
```

File: src/pipe.toit [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+/**
+Returns the exit value of the process which can then be decoded into
+  exit code or signal number.
+
+See $exit_code and $exit_signal.
+*/
+wait_for subprocess:
+  wait_for_ subprocess
+  state := monitor.ResourceState_ process_resource_group_ subprocess
```
Comment (kasperl) [inline src/pipe.toit:248]: Dispose the state after calling wait.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/pipe.toit (hunk 1 above)
Comment (floitsch) [reply src/pipe.toit:81]: https://github.com/toitlang/pkg-host/pull/2

File: src/pipe.toit (hunk 2 above)
Comment (floitsch) [reply src/pipe.toit:248]: https://github.com/toitlang/pkg-host/pull/2

**Pull request merged**